### PR TITLE
PIN-3997 Implemented update of the Risk Analysis

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It inherit from [PIN-3994](https://github.com/pagopa/interop-be-catalog-management/pull/226)

[PIN-3994]: https://pagopa.atlassian.net/browse/PIN-3994?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -832,7 +832,59 @@ paths:
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/RiskAnalysisSeed'  
+              $ref: '#/components/schemas/RiskAnalysisSeed'
+  /eservices/{eServiceId}/riskanalysis/{riskAnalysisId}/:
```
Comment (beetlecrunch) [inline src/main/resources/interface-specification.yml]:
```suggestion
  /eservices/{eServiceId}/riskanalysis/{riskAnalysisId}:
```

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/serializer/v1/package.scala [hunk 2]
```diff
@@ -371,4 +371,22 @@ package object v1 {
       catalogItem <- convertCatalogItemsFromV1(event.catalogItem)
     } yield CatalogItemRiskAnalysisAdded(catalogItem = catalogItem, catalogRiskAnalysisId = event.catalogRiskAnalysisId)
 
+  implicit def catalogItemRiskAnalysisUpdatedV1PersistEventSerializer
+    : PersistEventSerializer[CatalogItemRiskAnalysisUpdated, CatalogItemRiskAnalysisUpdatedV1] = event =>
+    for {
+      catalogItem <- convertCatalogItemsToV1(event.catalogItem)
+    } yield CatalogItemRiskAnalysisUpdatedV1(
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/serializer/v1/package.scala]: If Antonio were with us...

File: src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/serializer/v1/package.scala [hunk 3]
```diff
@@ -371,4 +371,22 @@ package object v1 {
...
+      catalogItem = catalogItem,
+      catalogRiskAnalysisId = event.catalogRiskAnalysisId
+    )
+
+  implicit def catalogItemRiskAnalysisUpdatedV1PersistEventDeserializer
+    : PersistEventDeserializer[CatalogItemRiskAnalysisUpdatedV1, CatalogItemRiskAnalysisUpdated] = event =>
+    for {
+      catalogItem <- convertCatalogItemsFromV1(event.catalogItem)
+    } yield CatalogItemRiskAnalysisUpdated(
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/serializer/v1/package.scala]: same

File: src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogPersistentBehavior.scala [hunk 4]
```diff
@@ -121,6 +121,23 @@ object CatalogPersistentBehavior {
...
+          .map { eService =>
+            val updatedItem = eService
+              .copy(riskAnalysis = eService.riskAnalysis.filter(_.id.toString != riskAnalysis.id) :+ riskAnalysis)
+
+            Effect
+              .persist(CatalogItemRiskAnalysisUpdated(updatedItem, riskAnalysis.id.toString))
+              .thenRun((_: State) => replyTo ! StatusReply.Success(Done))
+          }
+          .getOrElse {
+            replyTo ! StatusReply.error[Done](EServiceRiskAnalysisNotFound(eServiceId, riskAnalysis.id.toString))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogPersistentBehavior.scala]:
```suggestion
            replyTo ! StatusReply.error[Done](EServiceNotFound(eServiceId))
```

File: src/main/scala/it/pagopa/interop/catalogmanagement/error/CatalogManagementErrors.scala [hunk 5]
```diff
@@ -35,4 +35,7 @@ object CatalogManagementErrors {
 
   case object ElementNotFoundAfterUpdate extends ComponentError("0010", s"Element cannot be found after update")
 
+  final case class EServiceRiskAnalysisNotFound(eServiceId: String, riskAnalysisId: String)
+      extends ComponentError("0010", s"RiskAnalysis with id $riskAnalysisId of EService $eServiceId not found")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogmanagement/error/CatalogManagementErrors.scala]:
```suggestion
      extends ComponentError("0011", s"RiskAnalysis with id $riskAnalysisId of EService $eServiceId not found")
```

**Review by galales — COMMENTED** [review]
LGTM, just 2 very minor points

**Review by galales — APPROVED** [review]

**Pull request merged**